grid/chat/members/maps/slack.suma

```diff
@@ -0,0 +1,189 @@
+profile = "chat/members@1.0"
+provider = "slack"
+
+"""
+GetMembers map
+"""
+map GetMembers {
+  http GET "/users.list" {
+    request "application/x-www-form-urlencoded" {
+      query {
+        cursor = input.page
+        limit = input.limit || 100
+      }
+
+      headers {
+        "Authorization" = `Bearer ${parameters.accessToken}`
+      }
+    }
+
+    response 200 "application/json" {
+      error = call MapSlackError(
+        statusCode = statusCode,
+        error = body,
+        headers = headers,
+      ) if (!body.ok)
+
+      return map error if (!body.ok) error
+
+      members = call MapMembers(members = body.members)
+      rateLimit = call MapRateLimitDetails(headers = headers)
+      result.members = members
+
+      set if (body.response_metadata && body.response_metadata.next_cursor) {
+        result.nextPage = body.response_metadata.next_cursor
+      }
+
+      set if (rateLimit) {
+        result.rateLimit = rateLimit
+      }
+
+      map result result
+    }
+
+    response {
+      error = call MapSlackError(
+        statusCode = statusCode,
+        error = body,
+        headers = headers,
+      )
+      
+      map error error
+    }
+  }
+}
+
+"""
+Operation MapMembers
+Returns array mapping members in received body
+"""
+operation MapMembers {
+  members = []
+
+  members = args.members.map(member => {
+    const resultMember = {
+      id: member.id,
+      isBot: member.is_bot,
+    }
+
+    if (member.profile && member.profile.display_name) {
```

Comment:
Doesn't Slack expose username's for internal purposes? Or it shouldn't be used?

Reply:
Since slack does not really describe which to use, I'm using `name` as a fallback. It seems [here](https://api.slack.com/changelog/2017-09-the-one-about-usernames) that they prefer to use `display_name` so I'm not sure.

Reply:
Ok I found better description of user object and they do not recommend using `name` field ( see [here](https://api.slack.com/types/user) )